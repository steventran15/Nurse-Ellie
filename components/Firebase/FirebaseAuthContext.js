import React, { createContext } from 'react';

export const FirebaseAuthContext = createContext();