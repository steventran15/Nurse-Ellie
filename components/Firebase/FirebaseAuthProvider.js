import React, { useState, useEffect } from 'react';
import { firebase } from './config';
import { FirebaseAuthContext } from './FirebaseAuthContext';

const FirebaseAuthProvider = ({ children }) => {

    const [currentUser, setUser] = useState(null);

    useEffect(() => {
        let unsubscribe = firebase.auth().onAuthStateChanged(user => {
            if (user) {
                setUser(user);
            }
        })

        return () => {
            unsubscribe();
        }

    }, []);

    return (

        <FirebaseAuthContext.Provider value={{ currentUser }}>
            {children}
        </FirebaseAuthContext.Provider>

    );
};

export {
    FirebaseAuthProvider
}