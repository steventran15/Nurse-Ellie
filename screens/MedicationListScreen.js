import React, { useState, useEffect, useContext} from 'react';
import { View, Text, TextInput, KeyboardAvoidingView, TouchableOpacity, FlatList, Button, Dimensions, StyleSheet, Alert } from 'react-native';
import { firebase } from '../components/Firebase/config';

import * as Animatable from 'react-native-animatable';

import MedIconIndex from '../components/MedicationImages';

import PatientStyles from '../styles/PatientStyleSheet';
import Background from '../components/background';
import MedicationCard from '../components/MedicationCard';
import MenuIcon from '../assets/images/menu-icon.svg';
import MedicationsIcon from '../assets/images/medications-icon.svg';
import SearchIcon from '../assets/images/search-icon.svg';

import { FirebaseAuthContext } from '../components/Firebase/FirebaseAuthContext';
import * as fsFn  from '../utils/firestore';
import { ActivityIndicator } from 'react-native-paper';

const MedicationListScreen = ({navigation}) => {
    const { currentUser } = useContext(FirebaseAuthContext);
    const [medications, setMedications] = useState ([
        {medicationName: 'Monopril', function: 'High Blood Pressure', frequency: '1x/day', alert: '10:00AM', key: '1'}, 
        {medicationName: 'Cymbalta', function: 'Joint Pain', frequency: '1x/day', alert: '9:00AM', key: '2'}, 
        {medicationName: 'Codeine', function: 'Cough & Cold', frequency: '15ml/day', alert: '10:00AM', key: '3'},
    ])
    const [newMedications, setNewMedications] = useState([]);
    const [loading,setLoading] = useState(true);

    useEffect(() => {
        //collection listener, initializes local state with all user medication
        const listener = firebase.firestore()
            .collection("users")
            .doc(currentUser.uid)
            .collection("medications")
            .onSnapshot(querySnapshot => {
                const meds = [];
                querySnapshot.forEach(documentSnapshot =>{
                    let id = documentSnapshot.id;
                    let data = documentSnapshot.data();
                    meds.push({
                        'docId' : id,
                        'medication': data,
                    })
                });
                setNewMedications(meds);
                setLoading(false);
            });
        // unsubscribe from events when no longer in use
        //return () => listener();
    }, []);


    return (
        <KeyboardAvoidingView style={PatientStyles.background} behaviour="padding" enabled>
            <Background/>
            <TouchableOpacity style={PatientStyles.menuButton} onPress={()=> navigation.openDrawer()}>
                <MenuIcon/>
            </TouchableOpacity>
            <Animatable.View style={PatientStyles.drawer} animation="fadeInUpBig"> 
                <View style={PatientStyles.header}>
                    <Text style={PatientStyles.title}>
                        Medications
                    </Text>
                    <MedicationsIcon/>
                </View>
                <View style={styles.searchInput}>
                    <TextInput style={{minWidth: screenWidth*0.8}}placeholder="Search Medication..."></TextInput>
                    <TouchableOpacity onPress={()=>Alert.alert('searching medication...')}>
                        <SearchIcon/>
                    </TouchableOpacity>
                </View>
                { loading ? (
                    <View style={{padding:screenHeight *.5}}>
                        <ActivityIndicator/>
                    </View>
                ) : (
                    <FlatList 
                        data={newMedications.sort( (a,b) => {
                            return a.medication.nameDisplay.localeCompare(b.medication.nameDisplay);
                        })} 
                        keyExtractor={(item) => item.docId}
                        renderItem={({item}) => (
                        <TouchableOpacity style={styles.searchButton} onPress={()=>navigation.navigate('Medication', {item: item})}>
                            <MedicationCard>
                                <View style={{justifyContent:'center', paddingHorizontal:6, width: 60}}>
                                    {MedIconIndex.index[item.medication.medIcon]}
                                </View>
                                <View style={styles.medicationInfoView}>
                                <Text style={styles.medicationFont}>{item.medication.nameDisplay}</Text>
                                {/*  <Text style={styles.functionFont}>no simple function in db yet</Text> */}
                                <Text style={styles.frequencyfont}>{item.medication.strength}</Text>
                                </View>
                                <View style={styles.timeView}>
                                    <Text style={styles.timeFont}>{item.medication.intakeTime}</Text>
                                </View>
                            </MedicationCard>
                        </TouchableOpacity>
                    )}/>
                )}
                <Button title="ADD NEW MEDICATION" color='#42C86A' onPress={()=>{navigation.navigate('AddMedication'); console.log(JSON.stringify(newMedications[0]));}}/>
            </Animatable.View>
        </KeyboardAvoidingView>
    )
}

var screenHeight = Dimensions.get("window").height;
var screenWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
    medicationFont: {
        fontFamily: 'roboto-regular', 
        fontSize: 20, 
        color: 'rgba(0, 0, 0, 0.85)'
    },
    functionFont:{
        fontFamily: 'roboto-regular', 
        fontSize: 14, 
        color: 'rgba(0, 0, 0, 0.7)'
    },
    frequencyfont: {
        fontFamily: 'roboto-regular', 
        fontSize: 12, 
        color: 'rgba(0, 0, 0, 0.7)'
    },
    timeFont: {
        fontFamily: 'roboto-regular', 
        fontSize: 18, 
        color: 'rgba(0, 0, 0, 0.85)'
    },
    medicationInfoView: {
        width: 170,
        paddingHorizontal: 10
    },
    timeView:{
        borderLeftColor: 'rgba(0, 0, 0, 0.33)', 
        borderLeftWidth: 1,
        paddingHorizontal: 15, 
        justifyContent: 'center', 
        width: 100
    },
    searchInput:{
        flexDirection: 'row',
        borderBottomColor: 'rgba(0, 0, 0, 0.6)',
        borderBottomWidth: 1, 
        minWidth: screenWidth * 0.80, 
        marginVertical:10
    },
    searchButton:{
        right: 5
    },
});

export default MedicationListScreen;