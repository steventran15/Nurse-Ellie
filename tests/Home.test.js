import React from 'react';
import renderer from 'react-test-renderer';

import { FirebaseAuthContext } from '../components/Firebase/FirebaseAuthContext';
import MedicationListScreen from '../screens/MedicationListScreen';

import { firebase } from '../components/Firebase/config';

const navigationProp = {
    navigate: jest.fn(),
    openDrawer: jest.fn(),
};

const firestoreMock = {
    collection: jest.fn().mockReturnThis(),
    doc: jest.fn().mockReturnThis(),
    onSnapshot: jest.fn().mockResolvedValueOnce(),
  };

export const FirebaseAuthProvider = ({ children }) => (
    <FirebaseAuthContext.Provider value={{ currentUser: { uid: 'test' } }}>
        {children}
    </FirebaseAuthContext.Provider>
);

test('Home Screen renders correctly', async () => {
    const wrapper = renderer.create(
        <FirebaseAuthProvider>
            <MedicationListScreen navigation={navigationProp} />
        </FirebaseAuthProvider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});