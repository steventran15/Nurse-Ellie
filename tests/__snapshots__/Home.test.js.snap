// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View
  behaviour="padding"
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#42C86A",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#42C86A",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(62, 185, 99, 0.8)",
          "borderRadius": 250,
          "height": 500,
          "position": "absolute",
          "right": -375,
          "top": 667,
          "width": 500,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 30,
          "height": 60,
          "left": 10,
          "position": "absolute",
          "top": 30,
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 125,
          "height": 250,
          "left": 150,
          "position": "absolute",
          "top": 222.33333333333334,
          "width": 250,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 300,
          "height": 600,
          "left": 300,
          "position": "absolute",
          "top": -333.5,
          "width": 600,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 175,
          "height": 350,
          "left": -250,
          "position": "absolute",
          "top": 667,
          "width": 350,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 30,
        "top": 40,
      }
    }
  >
    <svg
      data-file-name="SvgMenuIcon"
    />
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 4,
        "height": 1133.8999999999999,
        "opacity": 0,
        "paddingHorizontal": 30,
        "paddingVertical": 50,
        "position": "absolute",
        "top": 200.1,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "roboto-regular",
            "fontSize": 24,
            "fontWeight": "100",
          }
        }
      >
        Medications
      </Text>
      <svg
        data-file-name="SvgMedicationsIcon"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(0, 0, 0, 0.6)",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "marginVertical": 10,
          "minWidth": 600,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="Search Your Medications..."
        rejectResponderTermination={true}
        style={
          Object {
            "minWidth": 600,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <svg
          data-file-name="SvgSearchIcon"
        />
      </View>
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#42C86A",
              },
            ]
          }
        >
          ADD NEW MEDICATION
        </Text>
      </View>
    </View>
  </View>
</View>
`;
